scripts/upload-balsamic-housekeeper.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+import json
+from cg.store import Store, models
+from cg.apps.hk import HousekeeperAPI
+import datetime as dt
+from ruamel import yaml
+import click
+
+
+@click.command()
+@click.option("--database", help="StatusDB address")
+@click.option("--housekeeper-root", help="Housekeeper root path")
+@click.option("--housekeeper-db", help="Housekeeper database address")
+@click.option("--report-dir", help="Path where housekeeper reports were stored")
+@click.option("--case-dir", help="Path where all cancer cases are stored")
+def upload_cases(database, housekeeper_root, housekeeper_db, report_dir, case_dir):
+    """One-time script to upload all BALSAMIC output files to Housekeeper and StatusDB"""
+    store_api = Store(database)
+    housekeeper_config = {"housekeeper": {"root": housekeeper_root, "database": housekeeper_db}}
+    housekeeper_api = HousekeeperAPI(housekeeper_config)
+
+    balsamic_families = (
+        store_api.Family.query.outerjoin(models.Analysis)
+        .join(models.Family.links, models.FamilySample.sample)
+        .filter(models.Sample.data_analysis.ilike("%Balsamic%"))
+    )
+    fam_names = {x.internal_id: None for x in balsamic_families}
+    for obj in Path(case_dir).iterdir():
+        if obj.is_dir() and any(x in obj.name for x in fam_names):
+            for obj2 in obj.iterdir():
+                if obj2.as_posix().endswith(".json"):
+                    config_dict = dict(json.load(open(obj2, "r")))
+                    case_id = config_dict.get("analysis").get("case_id")
+                    creation_date = config_dict.get("analysis").get("config_creation_date")
+                    if creation_date:
+                        creation_date = dt.datetime.strptime(creation_date, "%Y-%m-%d %H:%M")
+                    if not case_id:
+                        case_id = obj.name
+                    case_id = case_id.split("_")[0]
+                    if not Path(f"{case_dir}/{obj.name}/analysis/bam/tumor.merged.bam").is_file():
+                        continue
+                    if fam_names.get(case_id):
+                        if fam_names.get(case_id).get("date") > creation_date:
+                            continue
+                    fam_names[case_id] = {
+                        "bv": config_dict.get("analysis").get("BALSAMIC_version"),
+                        "config_name": f"{case_dir}/{obj.name}/{obj2.name}",
+                        "date": creation_date,
+                        "path": obj.name,
+                        "hk": Path(report_dir, obj.name + ".hk"),
+                    }
+    for case_id in fam_names:
+        items = fam_names.get(case_id)
+        if items:
```

Comment:
```suggestion
        if not items:
            continue
```
And then you can move the following try-except block one indentation to the left making it more flat